alma/sfcc-cartridge
# Add partial capture in deferred capture

## Reason for change



[Linear task](https://linear.app/almapay/issue/MPP-648/add-partial-capture-to-deferred-capture)
We need to implement the possibility to make a partial capture in the deferred capture feature.

### Code changes



- We add a status PartialCaptured in field order.custom.ALMA_Deferred_Capture_Status.
- We add a field order.custom.ALMA_Deferred_Capture_Partial_Amount.
- If ALMA_Deferred_Capture_Partial_Amount is filled, the job fill the amount with this value for the capture. 
- We create lot of variables to simplify the capture code.
- We rectify and add unit tests.

### How to test

On the SFCC dev env, create 2 orders.
In the bo go to Merchant tools > Ordering > Orders
In one order, go to the Attributes tab and fill the Amount for partial capture field.
Run the job CapturePaymentOrders (Administration > Operations > Jobs)
When it’s completed, return on the 2 last orders and, in the Attributes tab, check this two fields: Capture Status and Amount captured



### Checklist for authors and reviewers




### Non applicable
